File: opspilot-enterprise/packages/shared-schema/src/opspilot_schema/policy.py
from __future__ import annotations
from typing import Optional, Any
from pydantic import BaseModel


class OpsPolicy(BaseModel):
    id: str
    name: str
    description: str
    type: str
    status: str
    effect: str
    scope: list[str] = []
    conditions: dict[str, Any] = {}
    created_at: str
    updated_at: str
    hit_count: int = 0
    last_hit_at: Optional[str] = None
    author: str
    version: str
    rego_snippet: Optional[str] = None


class PolicyHitRecord(BaseModel):
    id: str
    policy_id: str
    policy_name: str
    effect: str
    actor: str
    tool_name: str
    resource: str
    outcome: str
    timestamp: str
    trace_id: str

File: packages/shared-types/src/policy.ts
export type PolicyType = "approval_gate" | "rate_limit" | "scope_guard" | "time_window" | "risk_threshold" | "audit_only";
export type PolicyStatus = "active" | "inactive" | "draft";
export type PolicyEffect = "allow" | "deny" | "require_approval" | "alert_only";

export interface OpsPolicy {
  id: string;
  name: string;
  description: string;
  type: PolicyType;
  status: PolicyStatus;
  effect: PolicyEffect;
  scope: string[];
  conditions: Record<string, unknown>;
  created_at: string;
  updated_at: string;
  hit_count: number;
  last_hit_at: string | null;
  author: string;
  version: string;
  rego_snippet: string | null;
}

export interface PolicyHitRecord {
  id: string;
  policy_id: string;
  policy_name: string;
  effect: PolicyEffect;
  actor: string;
  tool_name: string;
  resource: string;
  outcome: "blocked" | "allowed" | "escalated";
  timestamp: string;
  trace_id: string;
}
